jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/NodeInstanceImpl.java

```diff
@@ -176,15 +175,7 @@ public final void trigger(NodeInstance from, String type) {
     	}
     	
     	if (from != null) {
-    	    int level = ((org.jbpm.workflow.instance.NodeInstance)from).getLevel();
-    	    ((org.jbpm.workflow.instance.NodeInstanceContainer)getNodeInstanceContainer()).setCurrentLevel(level);
-	    	Collection<Connection> incoming = getNode().getIncomingConnections(type);
-	    	for (Connection conn : incoming) {
-	    	    if (conn.getFrom().getId() == from.getNodeId()) {
-	    	        this.metaData.put("IncomingConnection", conn.getMetaData().get("UniqueId"));
-	    	        break;
-	    	    }
-	    	}
+            this.metaData.put("IncomingConnection", from.getNode().getMetaData().get("UniqueId"));
```

Comment:
"UniqueId" could be defined as constant (as it is used three times in this class) (SonarCloud code smell)

Reply:
ok